contrib/babelfishpg_tsql/src/hooks.c

```diff
@@ -1180,6 +1191,133 @@ tle_name_comparison(const char *tlename, const char *identifier)
 		return (0 == strcmp(tlename, identifier));
 }
 
+/*
+* A special case of the get_object_address_relobject() function, specifically
+* for the case of triggers in tsql dialect. We add a pg_trigger lookup to search
+* for the relation that the trigger is associated with, since the relation name
+* is not supplied by the user, and thus not a part of the *object list.
+*/
+static ObjectAddress
+get_trigger_object_address(List *object, Relation *relp, bool missing_ok)
+{
+	ObjectAddress address;
+	Relation	relation = NULL;
+	const char *depname;
+	Oid			reloid;
+	Relation	tgrel;
+	ScanKeyData		key;
+	SysScanDesc tgscan;
+	HeapTuple	tuple;
+	const char	*trigger_physical_schema = NULL;
+	const char	*trigger_logical_schema = NULL;
+	const char	*cur_dbo_physical_schema= NULL;
+	bool 	found_trigger = false;
+	const char	*pg_trigger_physical_schema = NULL;
+	const char	*pg_trigger_logical_schema = NULL;
+	const char	*cur_physical_schema = NULL;
+
+	if (sql_dialect != SQL_DIALECT_TSQL)
```

Comment:
Left a comment on engine PR as well - I would suggest removing this check from here since we are unnecessarily returning random values here. 

It would make more sense to check for sql dialect before invoking the hook.